libsolidity/ast/Types.h

```diff
@@ -1373,6 +1396,9 @@ class MappingType: public Type
 	Type const* keyType() const { return m_keyType; }
 	Type const* valueType() const { return m_valueType; }
 
+protected:
+	std::vector<Type const*> decomposition() const override { return {m_valueType}; }
```

Comment:
It may be a bit confusing and misleading that the decomposition of a mapping type is only it's value type, but doesn't involve the key type... but maybe it's fine...

Reply:
yeah, the key is a super type, the value is the sub type ;)

Reply:
Hm... I'm not sure about that :-D. But I guess no matter what we do, it will need explicit comments and care...

Reply:
Do we store keys or just use them to calculate hashes? (In the former case, we should add `keyType` to subtypes, perhaps, but in the later case we should not.)

Reply:
I don't think we store them. That's the reason why you can't iterate over a mapping (unless you store the keys separately).

Reply:
No, they're not stored :-). But at least syntactically it's kind of part of the type still :-)... not sure what I would expect if I looked at a function "subTypes()" or "typeDecomposition()" of a type ``mapping(typeA => typeB)``... but yeah - there's probably no perfect name that makes this absolutely and entirely clear, so at least I'm fine with any name plus a few documentation comments explaining what it does especially for mappings.

Reply:
Yeah... Both `subtypes()` and `decomposition()` sound OK to me, although after my own back-and-forth I might prefer `subtypes()` for now.

But let's do a mini-poll. What would be your preference?

Reply:
`typesYouCanSplitSerializedRepresentationOfThisTypeInto()`. Or `smash()` for short :)

But seriously `decomposition()` is fine in my opinion so I'd vote for that.

Also, another idea would be simply `parts()`.

Reply:
decomposition sounds like it captures the full extent of the type, i.e. if you decompose the type, you can re-compose and get the same thing - which is not true.

Reply:
``storedSubTypes()`` :-). Well, I don't have any preference, as long as there's a big fat comment for it.

Reply:
I suggested some comments below. Not sure if fat enough though :)